### PR TITLE
fix(api): init wf without repowebhook on existing workflow

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: engine/api/workflow/hook.go [hunk 1]
```diff
@@ -138,7 +138,7 @@ func hookRegistration(ctx context.Context, db gorp.SqlExecutor, store cache.Stor
 		}
 		if h.HookModelName == sdk.RepositoryWebHookModelName || h.HookModelName == sdk.GitPollerModelName || h.HookModelName == sdk.GerritHookModelName {
 			if wf.WorkflowData.Node.Context.ApplicationID == 0 || wf.Applications[wf.WorkflowData.Node.Context.ApplicationID].RepositoryFullname == "" || wf.Applications[wf.WorkflowData.Node.Context.ApplicationID].VCSServer == "" {
-				return sdk.NewErrorFrom(sdk.ErrForbidden, "cannot create a git poller or repository webhook on an application without a repository")
+				return sdk.NewErrorFrom(sdk.ErrForbidden, "cannot create a git poller or repository webhook on an application without a repository: %+v", wf.Applications[wf.WorkflowData.Node.Context.ApplicationID])
```
Comment (yesnault) [inline engine/api/workflow/hook.go]:
```suggestion
				return sdk.NewErrorFrom(sdk.ErrForbidden, "cannot create a git poller or repository webhook on an application without a repository")
```

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#10484.0 ✘  
* Build minimal
  * cdsctl linux ✘  
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#10484.1 ✘  
* Build minimal
  * cdsctl linux ✘  
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#10497.1 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#10499.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔ 
* Build other os/arch
  * cdsctl freebsd windows amd64 -   
  * cdsctl openbsd amd64 -   
  * Doc Build -   
  * Engine Build darwin amd64 -   
  * Engine Build freebsd amd64 -   
  * Engine Build linux arm/arm64 -   
  * Engine Build openbsd amd64 -   
  * Engine Build windows amd64 -   
  * Plugins darwin amd64 -   
  * Plugins freebsd -   
  * Plugins integrations K8S -   
  * Plugins linux amd64 -   
  * Plugins linux arm -   
  * Plugins linux arm64 -   
  * Plugins openbsd -   
  * Plugins windows amd64 -   
  * Worker darwin / windows / freebsd / openbsd -   
  * Worker linux arm -   
* Package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔

**Pull request merged**